modules/dnn/src/tflite/tflite_importer.cpp

```diff
@@ -219,42 +231,70 @@ void TFLiteImporter::populateNet()
                 Mat data = allTensors[op_inputs->Get(0)];
-                Mat dataFP32;
-                convertFp16(data, dataFP32);
-                allTensors[op_outputs->Get(0)] = dataFP32;
-                continue;
+                if (!data.empty()) {
+                    // Dequantize a buffer
+                    Mat dataFP32;
+                    convertFp16(data, dataFP32);
+                    allTensors[op_outputs->Get(0)] = dataFP32;
+                    continue;
+                }
             }
 
             DispatchMap::const_iterator iter = dispatch.find(type);
             if (iter == dispatch.end())
                 CV_Error(Error::StsNotImplemented, "Unsupported operator type " + type);
 
-            CALL_MEMBER_FN(*this, iter->second)(*op, type, layerParams);
+            CALL_MEMBER_FN(*this, iter->second)(*op, type, layerParams, activParams);
 
             // Collect input blobs
-            std::vector<int> layerInputs;
-            std::vector<DataLayout> inpLayouts;
-            for (int idx : *op_inputs) {
-                if (layerIds.find(idx) != layerIds.end()) {
-                    layerInputs.push_back(idx);
-                    inpLayouts.push_back(layouts[idx]);
-                    continue;  // Output from a different layer
+            if (layerParams.blobs.empty()) {
+                for (int idx : *op_inputs) {
+                    if (layerIds.find(idx) != layerIds.end()) {
+                        continue;  // Output from a different layer
+                    }
+                    Mat blob = allTensors[idx];
+                    layerParams.blobs.push_back(blob.u ? blob : blob.clone());  // some tensors are owned by OpenCV
                 }
-
-                Mat blob = allTensors[idx];
-                layerParams.blobs.push_back(blob.u ? blob : blob.clone());  // some tensors are owned by OpenCV
             }
 
-            int layerId = dstNet.addLayer(layerParams.name, layerParams.type, layerParams);
+            int dtype = CV_32F;
+            if (isInt8(*op)) {
+                dtype = CV_8S;
+                if (layerParams.type != "Quantize")
+                    layerParams.type += "Int8";
+
+                if (!layerParams.has("zeropoints")) {
+                    float inpScale, outScale;
+                    int inpZero, outZero;
+                    getQuantParams(*op, inpScale, inpZero, outScale, outZero);
+
+                    layerParams.set("input_scale", inpScale);
+                    layerParams.set("input_zeropoint", inpZero);
+                    layerParams.set("scales", outScale);
+                    layerParams.set("zeropoints", outZero);
+                }
+            }
+            int layerId = dstNet.addLayer(layerParams.name, layerParams.type, dtype, layerParams);
 
             // Connect layer to inputs
             int i = 0;
-            for (int idx : layerInputs) {
+            std::vector<DataLayout> inpLayouts;
```

Comment:
Problem of handlers API is here.
We should not have any heavy postprocessing after the handler call.

Post-processing should be done in the handler itself (because handler defines post-processing steps).
Provide a method for registration of DNN layers to be called from handlers.

See ONNX importer implementation for reference.

Reply:
Thanks! I reworked both fused activations logic and created an `addLayer` method. Here is a graph for EfficientDet I used to verify that activations were added correctly: [efficientdet.dot](https://github.com/opencv/opencv/files/11298016/efficientdet.dot.txt) ([efficientdet.pdf](https://github.com/opencv/opencv/files/11298014/efficientdet.pdf)). Unfortunately, for this model they work as identity function and even with unconnected activations, test passes.